frontend/src/wizards/Argo/SourceSelector.tsx

```diff
@@ -109,87 +74,35 @@ export function SourceSelector(props: SourceSelectorProps) {
         <Tile id="helm" value="Helm" label={t('Helm')} icon={<HelmIcon />} description={t('Use a Helm repository')} />
       </WizTiles>
       {/* Git repo */}
-      <WizHidden hidden={(data) => data.spec.template.spec.source.path === undefined}>
-        <GitURLPath
-          t={t}
-          channels={channels}
-          getGitRevisions={getGitRevisions}
-          gitChannels={gitChannels}
-          path={'spec.template.spec.source.repoURL'}
-          setCreatedChannels={setCreatedChannels}
-          setGitRevisionsAsyncCallback={setGitRevisionsAsyncCallback}
-        />
-
-        <WizHidden hidden={(data) => data.spec.template.spec.source.repoURL === ''}>
-          <WizAsyncSelect
-            path="spec.template.spec.source.targetRevision"
-            label={t('Revision')}
-            labelHelp={t('Refer to a single commit')}
-            placeholder={t('Enter or select a tracking revision')}
-            asyncCallback={gitRevisionsAsyncCallback}
-            isCreatable
-            onValueChange={(value, item) => {
-              const channel = channels?.find(
-                (channel) => channel?.spec?.pathname === item.spec.template.spec.source.repoURL
-              )
-              const path = createdChannels.find((channel) => channel === item.spec.template.spec.source.repoURL)
-              setGitPathsAsyncCallback(
-                () => () =>
-                  getGitPathList(
-                    {
-                      metadata: {
-                        name: channel?.metadata?.name ?? '',
-                        namespace: channel?.metadata?.namespace ?? '',
-                      },
-                      spec: {
-                        pathname: channel?.spec.pathname ?? path ?? '',
-                        type: 'git',
-                      },
-                    },
-                    value as string,
-                    getGitPaths,
-                    item.spec.template.spec.source.repoURL
-                  )
-              )
-            }}
+      <ItemContext.Provider value={data.spec.template.spec.source}>
+        <WizHidden hidden={(data) => data.path === undefined}>
+          <RepoURL channels={gitChannels} />
+          <WizHidden hidden={(data) => data.repoURL === ''}>
+            <GitRevisionSelect channels={channels ?? []} />
+            <GitPathSelect channels={channels ?? []} />
+          </WizHidden>
+        </WizHidden>
+        {/* Helm repo */}
+        <WizHidden hidden={(data) => data.chart === undefined}>
+          <RepoURL channels={helmChannels} />
```

Comment:
```suggestion
      <ItemContext.Provider value={data?.spec?.template?.spec?.source}>
        <WizHidden hidden={(data) => data.path === undefined}>
          <RepoURL name="git" channels={gitChannels} />
          <WizHidden hidden={(data) => data.repoURL === ''}>
            <GitRevisionSelect channels={channels ?? []} />
            <GitPathSelect channels={channels ?? []} />
          </WizHidden>
        </WizHidden>
        {/* Helm repo */}
        <WizHidden hidden={(data) => data.chart === undefined}>
          <RepoURL name="helm" channels={helmChannels} />
```

Reply:
Linter was angry here too; name is a required prop here, apparently. This naming convention is found in other places in the related code so I used it here.